controllers/models/index.js

```diff
@@ -11,12 +11,45 @@ exports.retrieveEndpoints = () => {
     })
 }
 
-exports.retrieveTopics = () => {
-    return db.query('SELECT * FROM topics')
+const retrieveData = (sqlQuery) => {
+    return db.query(sqlQuery)
     .then((result) => {
         if(result.rows.length === 0) {
-            return Promise.reject({status: 404, message: 'nothing found'})
+            const dataName = result.fields[0].name.match(/^[a-z]+/i)[0]
+            return Promise.reject({status: 404, message: `${dataName} not found`})
         }
         return result
     })
+}
+
+exports.retrieveTopics = () => {
+    let sqlQuery = `SELECT * FROM topics`
+    return retrieveData(sqlQuery)
+}
+
+exports.retrieveArticlesById = (id) => {
+    let sqlQuery = `SELECT * FROM articles WHERE article_id=${id}`
+    sqlQuery += ` ORDER BY created_at DESC`
+    return retrieveData(sqlQuery)
+}
+
+exports.retrieveArticles = (id) => {
+    //COALESCE converts the value to 0 if its null
+    //AS comment_count sets the column name
+    //CAST(... AS INT) converts to an interger (apparently GROUP BY returns as a VARCHAR)
+    //COUNT(*) counts all the items in the group
+    //GROUP BY creates groups for all comments seperated by article_id
+    //SQL hurts
+    let sqlQuery = `SELECT author, title, articles.article_id, topic, created_at, votes, article_img_url, COALESCE(comments.comment_count, 0) AS comment_count FROM articles
```

Comment:
Same as above, const > let. Lets make sure we are sticking to good conventions here. Later we might want to allow this endpoint to take in queries from the user meaning we might need to make this query editable and make it a let, but right now we don't need to do that and don't have a test that has driven us to need to make this a let.